screens/clubDetailsPage.js

```diff
@@ -122,76 +130,221 @@ class clubDetailsPage extends Component{
             user = snapshot.toJSON();
             console.log('user: ',user);
             if (Object.keys(user).includes("clubs") && Object.keys(user.clubs).includes(clubId)) {
-                this.setState({clubMember: true})
+                this.setState({clubMember: true,
+                userInfo: user})
                 console.log('member of this club')
             }
             else {
-                this.setState({clubMember: false})
+                this.setState({clubMember: false,
+                userInfo: user})
                 console.log('NOT a member of this club')
             }
         }
-        this.setState({userInfo: user
-                    })
+        //this.setState({userInfo: user
+         //           })
         //console.log('userInfo: ',this.state.userInfo)
         });
     }
     
 
     groupButton() {
         const mem = this.state.clubMember
+        const admin = this.state.userInfo.admin
+        const clubAdmin = this.state.userInfo.clubAdminId
+        const currClub = this.state.clubId
         if (mem === true) {
             return (
                 <Button style={styles.button} mode="outlined" onPress = {this.leaveClub} > Leave This Club </Button>
             )
         }
+
+        else if (currClub === clubAdmin) {
+            return (null)
+        }
         else {
             return (<Button style={styles.button} mode="outlined" onPress = {this.joinClub}> Join This Club </Button>)
         }
     }
 
+    edit = () => {
+        this.setState({edit: true})
+    }
+
+    save = () => {
+        this.setState({edit: false})
+        const db = firebase.database().ref('/clubs/'+this.state.clubId)
+        const newDesc = this.state.newDescription
+        db.child('/description').set(newDesc)
+        this.setState({clubDesc: newDesc})
+    }
+    cancel = () => {
+        const oldDesc = this.state.clubDesc
+        this.setState({edit: false,
+                        newDescription: oldDesc})
+    }
+
+    editButton() {
+        const admin = this.state.userInfo.admin
+        const clubAdmin = this.state.userInfo.clubAdminId
+        const currClub = this.state.clubId
+        //console.log('clubAdminId: ',clubAdmin)
+        //console.log('currClub: ',currClub)
+        if (currClub === clubAdmin) {
+            return (
+                <Button style={styles.button} mode="outlined" onPress = {this.edit} > Edit Club Description</Button>
+            )
+        }
+        else {
+            return (null)
+        }
+    }
+
+    saveButton() {
```

Comment:
We can remove this function if it is not being used.

Reply:
deleted